mappings/net/minecraft/util/math/Quaternion.mapping

```diff
@@ -0,0 +1,31 @@
+CLASS net/minecraft/class_1158 net/minecraft/util/math/Quaternion
+	FIELD field_5656 components [F
+	METHOD <init> (FFFF)V
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+		ARG 4 w
+	METHOD <init> (FFFZ)V
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+		ARG 4 w
+	METHOD <init> (Lnet/minecraft/class_1158;)V
+		ARG 1 other
+	METHOD <init> (Lnet/minecraft/class_1160;FZ)V
+		ARG 1 direction
+		ARG 2 magnitude
+		ARG 3 normalize
```

Comment:
Something's weird about this, are you sure about `magnitude`? If the second argument is really the magnitude, then normalize doesn't make sense, since you could just pass a magnitude of `1` to the constructor to make it normal.

Reply:
Well `normalize` alters the magnitude, and magnitude is used to alter the x/y/z components.

        if (normalize) {
            magnitude *= 0.017453292f;
        }
        float float5 = sin(magnitude / 2.0f);
        (this.components = new float[4])[0] = direction.x() * float5;
        this.components[1] = direction.y() * float5;
        this.components[2] = direction.z() * float5;
        this.components[3] = cos(magnitude / 2.0f);

Maybe someone else can tell me more definitively what's going on here.

Reply:
"Normalize" means "make the magnitude 1". `0.017453292f` is `pi / 180`, so it seems like the `magnitude` is named wrongly, it's actually an angle. The third parameter is true if the second parameter should be in degrees.

Reply:
I was just thinking that, but couldn't figure out what `0.017453292f` was as a factor of Pi.

Reply:
Here's a description of what the constructor does: https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles#Definition

The first parameter is the direction of the axis of rotation, the second is the angle by which to rotate around that axis, and the third is whether that angle is given in degrees rather than radians.

I suggest the names `axis`, `angle`, `degrees`.

Reply:
Done.